src/packages/cli/src/args.ts

```diff
@@ -86,64 +151,20 @@ export default function (version: string, isDocker: boolean) {
           const state = {};
           for (const option in categoryObj) {
             const optionObj = categoryObj[option];
-            if (optionObj.disableInCLI !== true) {
-              const shortHand = [];
-              const legacyAliases = [];
-
-              let description = highlight(optionObj.cliDescription || "");
-              if (optionObj.cliAliases) {
-                optionObj.cliAliases.forEach(alias => {
-                  if (alias.length === 1) shortHand.push(alias);
-                  else legacyAliases.push(alias);
-                });
-                description = chalk`${description}${EOL}{dim deprecated aliases: ${legacyAliases
-                  .map(a => `--${a}`)
-                  .join(", ")}}`;
-              }
-
-              const generateDefaultDescription = () => {
-                // default sometimes requires a config, so we supply one
-                return (state[option] = optionObj.default
-                  ? optionObj.default(state).toString()
-                  : undefined);
-              };
-              const defaultDescription =
-                optionObj.defaultDescription || generateDefaultDescription();
-
-              // we need to specify the type of each array so yargs properly casts
-              // the types held within each array
-              const { cliType } = optionObj;
-              const array = cliType && cliType.startsWith("array:"); // e.g. array:string or array:number
-              const type = (array
-                ? cliType.slice(6) // remove the "array:" part
-                : cliType) as YargsPrimitiveCliTypeStrings;
-
-              const options: Options = {
-                group,
-                description,
-                alias: shortHand,
-                defaultDescription,
-                array,
-                type,
-                choices: optionObj.cliChoices,
-                coerce: optionObj.cliCoerce
-              };
-
-              const key = `${category}.${option}`;
-
-              // First, create *hidden* deprecated aliases...
-              flavorArgs = addAliases(flavorArgs, legacyAliases, key);
-
-              // and *then* create the main option, as options added later take precedence
-              // example: `-d --wallet.seed 123` is invalid (mutally exclusive). If aliases are defined _after_
-              // the main option definition the error message will be `Arguments deterministic and wallet.seed are mutually exclusive`
-              // when it should be `Arguments wallet.deterministic and wallet.seed are mutually exclusive`
-              flavorArgs = flavorArgs.option(key, options);
-            }
+            processOption(state, category, group, option, optionObj, flavorArgs);
           }
         }
 
-        // TODO: combine these cli options with core's `ServerOptions`
+        const state = {};
+        const categoryObj = serverDefaults.server;
+        const options = Object.keys(categoryObj);
```

Comment:
No need to get all the keys first since `categoryObj` is already iterable via `for(const option in categoryObj) {`. I'm not too worried about inherited properties showing up (which `Object.keys` omits).